localstack-core/localstack/services/lambda_/event_source_mapping/esm_worker.py

```diff
@@ -113,12 +117,7 @@ def poller_loop(self, *args, **kwargs):
         with self._state_lock:
             self.current_state = EsmState.ENABLED
             self.state_transition_reason = self.user_state_reason
-            # TODO: idea to not update store state but query state from esm_worker upon querying, but async deletion needs to modify store?!
-            # Update store state !?
-            # function_version = LambdaProvider._get_function_version_from_arn(self.esm_config["FunctionArn"])
-            # state = lambda_stores[function_version.id.account][function_version.id.region]
-            # esm_update = {"State": EsmState.ENABLED}
-            # state.event_source_mappings[self.esm_config["UUID"]].update(esm_update)
+
         while not self._shutdown_event.is_set():
```

Comment:
Do we reset the shutdown event on `start` somewhere? Otherwise a restart after enabling a disabled ESM won't last for long.

Reply:
Great catch 💯  Thank you for pointing this out.

Added the following to `start()` right before starting the poller thread:

```py
# Reset the shutdown event such that we don't stop immediately after a restart
self._shutdown_event.clear()
```